### PR TITLE
Add C implementation for `@stdlib/math/base/special/logit`

### LINKED ISSUE #762: [RFC]: Add C implementation for `@stdlib/math/base/special/logit`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/logit`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/logit).

```c
double stdlib_base_logit( const double p ); 
```

### Related Issues

#761.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #762.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/special/logit/src/main.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+* Evaluates the logit function.
+*
+* @param x    input value
+* @return	  output value
+*
+* @example
+* double out = stdlib_base_logit( 0.5 );
+* // returns ~0.405
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/src/main.c]: In the REPL,

```
In [1]: base.logit(0.5)
Out[1]: 0
```

So this return value does not look correct to me.

File: lib/node_modules/@stdlib/math/base/special/logit/test/test.native.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/test.native.js]:
```suggestion
var EPS = require( '@stdlib/constants/float64/eps' );
var abs = require( '@stdlib/math/base/special/abs' );
var tryRequire = require( '@stdlib/utils/try-require' );
```

File: lib/node_modules/@stdlib/math/base/special/logit/src/main.c (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/logit/src/main.c]: This is incorrect in `main.c`.

Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/src/main.c]:
```suggestion
* double out = stdlib_base_logit( 0.2 );
* // returns ~-1.386
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**